### ISSUE TITLE
middleware/kubernetes: fix response rewrite in autopath

### ISSUE THREAD

**Issue by chrisohaver** [coredns/coredns#761]
The RESPONSE option from autopath isn't working as expected.  DNS client behavior seems to be traversing the search path until data is returned, regardless of response code.  In effect, setting the response code does nothing.

**Comment by miekg** [comment]
Guessing it will continue until it see rcode.Success?
The less options to configure the better IMO

**Comment by chrisohaver** [comment]
The issue may be that I am not setting the rcode in the return correctly.
I am doing ...
```
wr := rewrite.NewResponseReverter(w, r)
...
rcode, nextErr := middleware.NextOrFailure(k.Name(), k.Next, ctx, wr, r)
...
rcode = dns.RcodeSuccess
r.Rcode = rcode
return rcode, nextErr
```
But the setting the rcode this way seems to have no effect on the rcode actually returned to the client, which receives NXDOMAIN.

**Comment by miekg** [comment]
`NewResponseReverter` will writte to the client, whatever you do afterwards will only make CoreDNS write *another* message (a SERVFAIL) or do nothing (assuming response was already written).

**Comment by chrisohaver** [comment]
Ah, I see how it works.  I think that I can create a new implementation of ResponseWriter, similar to ResponseReverter but also conditionally overwrites the response code based on the number of answers.

**Comment by chrisohaver** [comment]
cool - it works.  Need to test more, but may not need to remove the option after all ...

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] middleware/ku..." ]
>Ah, I see how it works.  I think that I can create a new implementation of ResponseWriter, similar to ResponseReverter but also conditionally overwrites the response code based on the number of answers.

That's sounds really dangerous... why can't you use the return code you already 
have?

**Comment by chrisohaver** [comment]
>why can't you use the return code you already have?

Because we need to change the response code from NXDOMAIN to NOERROR.  Thats the mechanism of the kludge... return a NOERROR, so the client doesn't continue searching the path.

**Comment by chrisohaver** [comment]
To clarify - the logic is not based on the number of answers as I said above... It's based on the return code... e.g. If the return code is NXDOMAIN, change it to NOERROR when writing it to the client.

... so yes, it uses the return code in the response writer to determine if the response code should be overridden with NOERROR.

**Comment by chrisohaver** [comment]
fixed in #771
